tcf_website/templates/reviews/new_review.html

```diff
@@ -31,5 +31,27 @@ <h1 class="h2 mt-2 mb-4">Write a review</h1>
   $(function () {
     $('[data-toggle="tooltip"]').tooltip()
   });
+
+  // Checks if required fields for review were filled
+  // If required fields are filled, timeout and hide the submit button so that the user can't spam click it
+  $(document).ready(function () {
+    $("#submitBtn").click(function () {
+        var reviewForm = document.getElementById("reviewForm");
+        // Check if required fields are filled
+        if (reviewForm.checkValidity()) {
+            var btn = document.getElementById("submitBtn");
+            btn.style.display = 'none';
```

Comment:
What do you think about simply disabling the button instead of hiding it entirely? Removing the element shifts the whole page, which is a little jarring, and I don't think I've ever seen something like this elsewhere. IMO an even better approach might just be to have some backend validation instead of disabling the frontend, but I don't want to invalidate the work you've done so far.

Reply:
Agreed, disabling the button is definitely better. However, when I tried doing that, it also disabled form submission and disabled the required field messages, so that if you clicked submit and didn’t fill out the required fields, nothing would happen and it wouldn’t tell you why. I can take out the frontend validation and simply opt for backend validation though.